gmail_yaml_filters/main.py

```diff
@@ -99,7 +101,10 @@ def main():
     credentials = get_gmail_credentials(client_secret_path=args.client_secret)
     gmail = get_gmail_service(credentials)
 
-    if args.action == 'upload':
+    if args.action == 'delete':
+        emptyruleset={}
+        prune_filters_not_in_ruleset(emptyruleset, service=gmail, dry_run=args.dry_run)
+    elif args.action == 'upload':
```

Comment:
The `--delete` command is a bit scary without any backup, but since we already have `--prune` I don't think it's any scarier than status quo. 😁  I'd merge that if it were a separate branch. I don't have a great way to write a regression test for that, though.

Reply:
I've implemented this as `--delete-all` in 9454797; it'll be part of the next release.

Reply:
This is more of my "rules in order" stuff. (#14)

Since you can't insert a changed rule you have to delete all the rules and re-add them to get them in order.